packages/components/src/internal/components/chart/HorizontalBarSection.tsx

```diff
@@ -1,100 +1,68 @@
 
 const DEFAULT_EMPTY_TEXT = 'No data available.';
 
 export interface HorizontalBarData {
-    percent: number;
-    count: number;
-    totalCount: number;
     backgroundColor?: string;
-    name?: string;
-    title: string;
-    href?: string;
     className?: string;
+    count: number;
     filled: boolean;
+    href?: string;
+    name?: string;
+    percent: number;
+    title: string;
+    totalCount: number;
 }
 
 interface Props {
-    title?: string;
-    subtitle?: React.ReactNode;
-    emptyText?: string;
     data: HorizontalBarData[];
+    emptyText?: string;
+    subtitle?: React.ReactNode;
+    title?: string;
 }
 
 export const HorizontalBarSection: FC<Props> = memo(props => {
     const { subtitle, title, data, emptyText } = props;
-    const horizontalBars = [];
+    let horizontalBars: ReactNode = <div className="horizontal-bar--empty-text">{emptyText ?? DEFAULT_EMPTY_TEXT}</div>;
 
     if (data?.length) {
-        let hasBegun = false;
-        const dataCount = data.length;
-        data.forEach((row, index) => {
-            if (row.percent > 0) {
-                const pct = row.percent;
-                const styleProps = {
-                    width: pct + '%',
-                };
-                if (row.backgroundColor) {
-                    styleProps['background'] = row.backgroundColor;
-                }
-                horizontalBars.push(
-                    <OverlayTrigger
-                        key={index}
-                        overlay={
-                            <Popover bsClass="popover" id="grid-cell-popover">
-                                {row.title}
-                            </Popover>
-                        }
-                        placement="top"
-                    >
+        horizontalBars = data
+            .filter(row => row.percent > 0)
+            .map((row, index) => {
+                const style: CSSProperties = { width: row.percent + '%', background: row.backgroundColor };
+                const overlay = (
+                    <Popover bsClass="popover" id="grid-cell-popover">
+                        {row.title}
+                    </Popover>
+                );
+                return (
+                    <OverlayTrigger key={index} overlay={overlay} placement="top">
                         <div
-                            key={index}
-                            style={styleProps}
+                            style={style}
                             data-title={row.title}
                             className={classNames('horizontal-bar-part', row.className, {
-                                'horizontal-bar--begin': !hasBegun,
-                                'horizontal-bar--filled': row.filled,
                                 'horizontal-bar--linked': !!row.href,
                                 'horizontal-bar--open': !row.filled || !row.backgroundColor,
-                                'horizontal-bar--end': index === dataCount - 1,
```

Comment:
Hmm.  I wonder when these classes went away. Perhaps when the refactor out of inventory happened.  Seems to be doing fine without them.